### PR TITLE
Load stix data into store

### PR DESCRIPTION
**Pull request by infosec-alchemist**
Created new docker container "load-stix" that will load the store with the initial Unfetter stix data.

### CODE REVIEW

File: load-stix/processor.js [hunk 1]
```diff
@@ -1,3 +1,10 @@
+var JSONAPISerializer = require('jsonapi-serializer').Serializer;
+
+var AttackPatternSerializer = new JSONAPISerializer('attack-patterns',{
```
Comment (exceptionfactory) [inline load-stix/processor.js]: This Serializer instance does not appear to be used and should be removed.

File: load-stix/processor.js [hunk 2]
```diff
@@ -16,17 +23,25 @@ const typePathMapping = {
     tools: "tools"
 };
 
-let port = 80;
+let port = 3000;
+if(process.argv.indexOf("-p")!=-1){
```
Comment (exceptionfactory) [inline load-stix/processor.js]: Check formatting settings to ensure that there is proper spacing for condition blocks.

File: load-stix/processor.js [hunk 3]
```diff
@@ -16,17 +23,25 @@ const typePathMapping = {
...
+    port = process.argv[process.argv.indexOf("-p")+1];
+}
+let host = "cti-stix-store";
+if(process.argv.indexOf("-h")!=-1){
+    host = process.argv[process.argv.indexOf("-h")+1];
+}
+/**
```
Comment (exceptionfactory) [inline load-stix/processor.js]: Remove commented portNumber checking if no longer used.

File: load-stix/processor.js [hunk 4]
```diff
@@ -80,6 +95,13 @@ function processBundleRecords(bundle) {
     }
 }
 
+function convertToJsonApi(inputJson){
```
Comment (exceptionfactory) [inline load-stix/processor.js]: This function does not appear to be used and should be removed.

File: load-stix/processor.js [hunk 5]
```diff
@@ -88,21 +110,30 @@ function processBundleRecords(bundle) {
...
     let processed = 0;
     let failed = 0;
+    var attributesList = [];
 
     records.forEach(function (record) {
+        for (let propertyKey in record){
+            attributesList.push(propertyKey);
+        }
+        var patternSerializer = new JSONAPISerializer(resourcePath,{
+            attributes: attributesList
```
Comment (exceptionfactory) [inline load-stix/processor.js]: The JSONAPISerializer also needs a keyForAttribute option set to underscore_case in order to maintain the correct attribute names for object properties.

File: load-stix/processor.js [hunk 6]
```diff
@@ -131,12 +162,20 @@ function postRecords(records, resourcePath) {
...
-        if (index > 1) {
-            processFile(val);
+
+function run(){
+    setTimeout(run2,3000);
+}
+
+function run2() {
+    console.log("Starting processor.js");
+    for (var i= process.argv.indexOf("-j")+1, len = process.argv.length; i< len;i++){
```
Comment (exceptionfactory) [inline load-stix/processor.js]: Specifying only the host command line option results in treating that argument as a file parameter.

File: load-stix/processor.js (hunk 1 above)
Comment (infosec-alchemist) [inline load-stix/processor.js]: removed

File: load-stix/processor.js (hunk 4 above)
Comment (infosec-alchemist) [inline load-stix/processor.js]: removed

File: load-stix/processor.js (hunk 5 above)
Comment (infosec-alchemist) [inline load-stix/processor.js]: added

File: load-stix/processor.js (hunk 6 above)
Comment (infosec-alchemist) [inline load-stix/processor.js]: -j is now required

**Comment by infosec-alchemist** [comment]
All review comments have been addressed

**Pull request merged**